### PR TITLE
Error handling, tests, lint, beat the patriarchy slightly

### PR DESCRIPTION
**Pull request by HairyFotr**

### CODE REVIEW

**Comment by Smotko** [comment]
QuantifiedCode found 3 issues that you can probably fix :)

**Comment by HairyFotr** [comment]
Will check out first warning type - maybe a separate helper class would be nice or something. The second type already has plenty of hits elsewhere - if you think it's a reasonable warning I will fix them all: https://www.quantifiedcode.com/app/project/gh:Psywerx:botko?groups=code_patterns%3A7I7lGdCx

**Comment by HairyFotr** [comment]
Green! (ignoring landscape, which is stuck)

Please test twitter, since I don't have api access, but for some reason pylint says:

    E: 47,15: Instance of 'APIMethod' has no 'user' member (no-member)
    E: 48,15: Instance of 'APIMethod' has no 'text' member (no-member)

**Comment by HairyFotr** [comment]
Oh, wait a second :) I'll put ready tag when done

**Comment by HairyFotr** [comment]
oh FFS, Quantified code - I didn't touch the signature :D

**Comment by HairyFotr** [comment]
Okay, I don't know how to get rid of these two. They're false positives IMO, also I didn't cause them, just touched their lines :)

And landscape (got a mail, it's not visible here for some reason) seems to think we've fixed almost everything, since it wraps pylint, and there's now a pretty lax .pylintrc here.

File: tests/test_plugins.py [hunk 1]
```diff
@@ -15,12 +22,134 @@
...
+        self._say_response('#psywerx', 'smotko', 'wow @all wow',
+                           '@msg off1 wow @all wow')
+        self._say_no_response('#psywerx', 'smotko', 'wow @all\' wow')
+
+    @patch("plugins.psywerx_groups.PsywerxGroups.request")
+    def test_basic_action(self, request):
+        resp = '_anything goes here_'
+        request.return_value = resp
+
+        for action in ['@mygroup', '@group', '@leaveall']:
```
Comment (HairyFotr) [inline tests/test_plugins.py]: Codacy Issue found: [Unused variable 'action'](https://www.codacy.com/app/hairyfotr/botko/file/2075810393/issues/source?bid=2114108&fileBranchId=3136536#l108)

File: src/plugins/read_links.py [hunk 2]
```diff
@@ -113,36 +113,34 @@ def _get_youtube_info(self, video_id):
...
                 t = t.find(".//title").text
                 t = t.strip().replace('\n', ' ')
                 if len(re.sub("[^a-zA-Z0-9]", "", t)) >= 5:
                     self.bot.say(random_response(WEB_RESPONSES) % {'title': t},
                                  channel)
-            except Exception as e:
+            except:
                 self.bot.log_error('Could not get title of webpage: "'
-                                   + msg + '" the exception was: ' + str(e))
+                                   + msg + '"')
```
Comment (HairyFotr) [inline src/plugins/read_links.py]: Codacy Issue found: [Unused argument 'nick'](https://www.codacy.com/app/hairyfotr/botko/file/2075810385/issues/source?bid=2114108&fileBranchId=3136536#l145)

File: src/plugins/psywerx_groups.py [hunk 3]
```diff
@@ -88,5 +88,5 @@ def _handleMentions(self, channel, nick, msg, line=None):
                          + " " + msg, channel)
 
     def handle_message(self, channel, nick, msg, line=None):
```
Comment (HairyFotr) [inline src/plugins/psywerx_groups.py]: Codacy Issue found: [Unused argument 'line'](https://www.codacy.com/app/hairyfotr/botko/file/2075810382/issues/source?bid=2114108&fileBranchId=3136536#l90)

**Pull request merged**